mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -68,6 +79,65 @@ class LogisticRegressionModel (
       case None => score
     }
   }
+
+  override def save(sc: SparkContext, path: String): Unit = {
+    val sqlContext = new SQLContext(sc)
+    import sqlContext._
```

Comment:
I tried that, and it fails to compile.  (I tried removed sqlContext from save(), as well as having it be an implicit val without the import.  Neither worked.)  Is there another import I need?
```
[error] /Users/josephkb/spark/mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala:89: type mismatch;
[error]  found   : org.apache.spark.rdd.RDD[org.apache.spark.mllib.classification.LogisticRegressionModel.Metadata]
[error]  required: org.apache.spark.sql.DataFrame
[error] Error occurred in an application involving default arguments.
[error]     val metadataRDD: DataFrame = sc.parallelize(Seq(metadata))
[error]                                                ^
[error] /Users/josephkb/spark/mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala:94: type mismatch;
[error]  found   : org.apache.spark.rdd.RDD[org.apache.spark.mllib.classification.LogisticRegressionModel.Data]
[error]  required: org.apache.spark.sql.DataFrame
[error] Error occurred in an application involving default arguments.
[error]     val dataRDD: DataFrame = sc.parallelize(Seq(data))
[error]                                            ^
[error] two errors found
[error] (mllib/compile:compile) Compilation failed
```
It may not be an issue though, with the other change you suggested below.  I'll see.